BrainPortal/lib/boutiques_input_copier.rb

```diff
@@ -0,0 +1,308 @@
+
+#
+# CBRAIN Project
+#
+# Copyright (C) 2008-2024
+# The Royal Institution for the Advancement of Learning
+# McGill University
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+#
+# Some tools modify the cached version of the input directory.
+#
+# This can lead to some issues when the input is used multiple times
+# on the same Bourreau and the cached version is already modified.
+#
+# To solve this issue, we can create a copy of the directory in the working
+# directory and use this copy in the command line.
+#
+#     apptool [DIRECTORY_INPUT] ...
+#
+# with the following input in Boutiques:
+#
+#     {
+#       "description": "A directory that contain the input data...",
+#       "id": "parent_input_id",
+#       "name": "Input",
+#       "optional": true,
+#       "type": "File",
+#       "command-line-flag": "-i",
+#       "value-key": "<-i input>"
+#     },
+#
+# and in the `cbrain:integrator_modules` section look like:
+#
+#     "BoutiquesInputCopier": {
+#       "parent_input_id": {"default": boolean, "hide": boolean, "description": string},
+#      },
+#
+# default:     boolean, true if the parent_input should be copied by default
+# hide:        boolean, true if the input should be hidden in the form
+# description: string, description of the input
+#
+# The final command line will be:
+#
+#     apptool <-i input_copy>...
+#
+#
+module BoutiquesInputCopier
+
+  # Note: to access the revision info of the module,
+  # you need to access the constant directly, the
+  # object method revision_info() won't work.
+  Revision_info=CbrainFileRevision[__FILE__] #:nodoc:
+
+
+  ############################################
+  # Portal Side Modifications
+  ############################################
+
+
+  # Utility Methods for the Module
+  ############################################
+
+  def copy_name(original_name) #:nodoc:
+    "inputcopier_tmp_" + original_name
```

Comment:
I am afraid this will break a bunch of tools.

If a tool requires an input with a particular naming convention, it will not work anymore. E.g. a tool that requires `sub-12345`.